packages/webpack/src/buildtime/aa.js

```diff
@@ -79,16 +79,21 @@ exports.generateIdentifierLookup = ({
   readableResourceIds,
 }) => {
   /**
-   * @typedef {Record<string, { aa: string; moduleId: number | string }>} PathMapping
+   * @typedef {Record<string, { aa: string; moduleIds: (number | string)[] }>} PathMapping
    */
   const pathsToIdentifiers = () => {
     /** @type {PathMapping} */
     const mapping = {}
     for (const p of paths) {
       if (p.path) {
-        mapping[p.path] = {
-          aa: getPackageNameForModulePath(canonicalNameMap, p.path),
-          moduleId: p.moduleId,
+        if (mapping[p.path]) {
+          mapping[p.path].moduleIds.push(p.moduleId)
+          diag.rawDebug(2, `Duplicated moduleId at path ${ p.path }, moduleIds: ${ mapping[p.path].moduleIds.join(', ') }`)
+        } else {
+          mapping[p.path] = {
+            aa: getPackageNameForModulePath(canonicalNameMap, p.path),
+            moduleIds: [p.moduleId],
+          }
         }
       }
```

Comment:
(suggestion) avoid nondescript variable name

```suggestion
  for (const {path, moduleId} of paths) {
    if (path) {
      if (mapping[path]) {
        mapping[path].moduleIds.push(moduleId)
        diag.rawDebug(2, `Duplicated moduleId at path ${ path }, moduleIds: ${ mapping[path].moduleIds.join(', ') }`)
       } else {
         mapping[path] = {
           aa: getPackageNameForModulePath(canonicalNameMap, path),
           moduleIds: [moduleId],
         }
       }
    }
```

Reply:
something is off with this suggestion diff, added the change manually.